### ISSUE TITLE
Test AdNauseam for Brave

### ISSUE THREAD

**Issue by cqx931** [dhowe/AdNauseam#1701]
Text ads from search engines are okay, but image ads are stopped by Brave's native ad blocking mechanism
<img width="1298" alt="Screenshot 2020-07-24 at 3 46 08 PM" src="https://user-images.githubusercontent.com/3056154/88429518-cd4ed080-cdc4-11ea-8213-d0724946a3c9.png">

**Comment by cqx931** [comment]
For ad collection/clicking to work, users need to go to `brave://settings/shields` and disable "Trackers & ads blocking"

**Comment by cqx931** [comment]
We can't know users brave settings using extension API, but we can add a warning message for brave users and a corresponding FAQ entry if we can verify in the content script that common ads(google ads) have been blocked.  @dhowe

**Comment by cqx931** [comment]
@dhowe

**Comment by dhowe** [comment]
So the content-script notices that common ads are blocked and sends a message to the core which triggers the extension warning ? How to recognize that a common ad is blocked ?

**Comment by cqx931** [comment]
Brave also block ads mostly by blocking web requests. If a user uses Brave and on initial install, we could potentially test it using one of our example pages. If adnauseam can't collect any ads from this page(or a dedicated test script to check whether iframes are fully rendered or not) while other potential reasons are eliminated, we can show a warning message related to Brave's native ad blocking.

<img width="684" alt="Screenshot 2020-08-18 at 11 27 11 AM" src="https://user-images.githubusercontent.com/3056154/90532888-c3ab5500-e145-11ea-96c4-65c6fcef9b02.png">

But I don't think this should be tested on every page, that would be too much work in vain.

**Comment by danielplatt** [comment]
How do I install AdNauseam for Brave? When dragging the .crx file into my browser it is automatically disabled and I see no way to enable it.

**Comment by dhowe** [comment]
@mneunomne pls test

**Comment by mneunomne** [comment]
Yes, that is indeed happening, can't turn the extension on. Not sure how it worked before, but apparently Brave is not allowing ads that are not from the store anymore:

> Extensions that have not been published on the Chrome or Brave Web Store are grayed out and you won't be able to turn them back on.

https://support.brave.com/hc/en-us/articles/360017914832-Why-am-I-seeing-the-message-extensions-disabled-by-Brave-?hl=en-GB

**Comment by dhowe** [comment]
Can you double-check the opera store as well?

**Comment by mneunomne** [comment]
@dhowe it Installs normally on Opera.

**Comment by dhowe** [comment]
you installed in brave via the opera store?

**Comment by mneunomne** [comment]
Ah ok, I understood your suggestion wrong before. No I cannot install it via opera store (but apparently I can't do that with any extension on Brave, did it use to work like that?)

In Brave browser:
![Screen Shot 2021-02-28 at 16 34 23](https://user-images.githubusercontent.com/4967860/109424253-3cd41080-79e3-11eb-842c-27904b3eb8ef.png)

**Comment by dhowe** [comment]
> apparently I can't do that with any extension on Brave, did it use to work like that?

Not sure -- but if they allow chrome store extensions in Brave, then worth a try

In any case the conclusion appears to be Brave only allows extensions from Brave/Chrome stores.

Here is some discussion about it: https://github.com/brave/brave-browser/issues/2457, containing a link to the relevant  Brave policy: https://support.brave.com/hc/en-us/articles/360017914832-Why-am-I-seeing-the-message-extensions-disabled-by-Brave-?hl=en-US

**Comment by dhowe** [comment]
The conclusion here appears to be that Brave no lonber allows extensions from outside the Brave or Chrome stores, and further, that this policy cannot be overridden by the user.

Here is some discussion about it: https://github.com/brave/brave-browser/issues/2457, containing a link to the relevant  Brave policy: https://support.brave.com/hc/en-us/articles/360017914832-Why-am-I-seeing-the-message-extensions-disabled-by-Brave-?hl=en-US

**Comment by divad42** [comment]
I was able to install AdNauseam on Brave from the unpacked zip on both Linux and Windows

**Comment by dhowe** [comment]
Excellent. If you can you include the steps to get them both installed, we can add to our wiki...

**Comment by georgebushfour20** [comment]
The chromium zip works on linux, the .crx file doesn't (extension cannot be enabled).

**Comment by dhowe** [comment]
@mneunomne can you update our page with instructions for installing Brave here: https://github.com/dhowe/AdNauseam/wiki/Install-AdNauseam-on-Chrome-Without-Google's-Permission

**Comment by caseysoftware** [comment]
On Brave (v1.22.71 on Mac), the quick fix:

- Visit the Extension config screen - brave://extensions/
- In the upper right, turn on Developer Mode which will make some buttons appear directly to the left
- Click the new button "Load Unpacked"
- Choose the directory where you expanded AdNauseam
- Assumed, not checked: You still have to keep the directory so put it somewhere to keep, not your Downloads folder

**Comment by dhowe** [comment]
@mneunomne pls update docs and close (lets also add a note in post-release testing to check the following suspicion from @caseysoftware):

> 
Brave may disable AdNauseam on upgrade. I've seen that with other no longer supported extensions.

**Comment by dhowe** [comment]
status @mneunomne ?

**Comment by mneunomne** [comment]
Sorry @dhowe somehow I missed the latests updates on this issue entirely.

It seems that yes, Brave as far as we are concerned, follows the same installation pattern than chrome. This information is already present in our documentation.

Regarding the image ads blocking, seems that they are being collected collected in general at a lesser level. For them to be collected normally, the user needs to change to "allow trackers and ads" configuration:

<img width="390" alt="Screenshot 2023-02-21 at 02 44 32" src="https://user-images.githubusercontent.com/4967860/220226688-490ae47e-d576-4c08-af57-121d1d0296d1.png">

**Comment by mneunomne** [comment]
Added this info on Brave installation segment on chromium installation wiki:

https://github.com/dhowe/AdNauseam/wiki/Install-AdNauseam-on-Chromium-based-browsers#optimal-configuration-for-ad-collection-on-brave

**Comment by mneunomne** [comment]
Currently AdNauseam works mostly fine with AdNauseam, although the ammout of collected ads will be less. Closing for now.
